From 44701675e1aeae1895ceb6e9db414e9818e4a3f3 Mon Sep 17 00:00:00 2001
From: Don Dutile <ddutile@redhat.com>
Date: Thu, 9 Nov 2017 06:50:24 -0500
Subject: [PATCH] [netdrv] qed: fix invalid use of sizeof in
 qed_alloc_qm_data()

Message-id: <20171109065423.8574-230-ddutile@redhat.com>
Patchwork-id: 197207
O-Subject: [PATCH RHEL-7.5 v2 229/468] qed: fix invalid use of sizeof in qed_alloc_qm_data()
Bugzilla: 1462433 1499362
RH-Acked-by: Jerry Snitselaar <jsnitsel@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

commit 2f7878c06e2d227aa5c405ddde356403b83e3509
Author: Wei Yongjun <weiyongjun1@huawei.com>
Date:   Tue Apr 25 07:07:18 2017 +0000

    qed: fix invalid use of sizeof in qed_alloc_qm_data()

    sizeof() when applied to a pointer typed expression gives the
    size of the pointer, not that of the pointed data.

    Fixes: b5a9ee7cf3be ("qed: Revise QM configuration")
    Signed-off-by: Wei Yongjun <weiyongjun1@huawei.com>
    Acked-by: Yuval Mintz <Yuval.Mintz@cavium.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1462433

Signed-off-by: Donald Dutile <ddutile@redhat.com>
Signed-off-by: Rafael Aquini <aquini@redhat.com>
---
 drivers/net/ethernet/qlogic/qed/qed_dev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/drivers/net/ethernet/qlogic/qed/qed_dev.c
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qed/qed_dev.c	2017-12-01 12:57:21.691058937 +0100
+++ src/drivers/net/ethernet/qlogic/qed/qed_dev.c	2017-12-01 12:57:32.802910576 +0100
@@ -820,7 +820,7 @@
 	if (!qm_info->qm_vport_params)
 		goto alloc_err;
 
-	qm_info->qm_port_params = kzalloc(sizeof(qm_info->qm_port_params) *
+	qm_info->qm_port_params = kzalloc(sizeof(*qm_info->qm_port_params) *
 					  p_hwfn->cdev->num_ports_in_engines,
 					  GFP_KERNEL);
 	if (!qm_info->qm_port_params)
